docs/topics/GATEWAY.md

```diff
@@ -216,7 +281,27 @@ The first step to establishing a gateway connection is to request a gateway URL
+When clients lose their connection to the gateway and are able to reconnect in a short period of time after, they can utilize a Gateway feature called "client resuming". Once reconnected to the gateway socket the client should send a [Gateway Resume](#DOCS_GATEWAY/gateway-resume) payload to the server. If successful, the gateway will respond by replaying all missed events to the client. Otherwise, the gateway will respond with an OP 9 (invalid session), in which case the client should send an OP 2 [Identify](#DOCS_GATEWAY/gateway-identify) payload to start a new connection. It is recommended that all Discord clients implement resume logic. The gateway can and will disconnect your websocket connection as it pleases without any warning. Implementing this feature will allow your client to reconnect seamlessly. Resuming is only supported when the amount of guilds in a given gateway connection is under 2,500 guilds. In order to resume as your bot grows, it's recommended that you [shard](#DOCS_GATEWAY/sharding) your gateway connection to reduce the number of guilds per gateway connection.
+
+### Disconnections
+
+If the gateway ever issues a disconnect to your client it will provide a close event code that you can use to properly handle the disconnection.
+
+###### Gateway Close Event Codes
+
+| Code | Description | Explanation |
+|------|-------------|-------------|
+| 4000 | unknown error | We're not sure what went wrong. Try reconnecting? |
+| 4001 | unknown opcode | You sent an invalid [Gateway OP Code](#DOCS_GATEWAY/gateway-op-codes). Don't do that! |
+| 4002 | decode error | You sent an invalid [payload](#DOCS_GATEWAY/sending-payloads) to us. Don't do that! |
+| 4003 | not authenticated | You sent us a payload prior to [identifying](#DOCS_GATEWAY/gateway-identify). |
+| 4004 | authentication failed | The account token sent with your [identify payload](#DOCS_GATEWAY/gateway-identify) is incorrect. |
+| 4005 | already authenticated | You sent more than one identify payloads. Don't do that! |
+| 4006 | session not valid | The session you tried to [resume](#DOCS_GATEWAY/resuming) was invalid. Reconnect and start a new one. |
```

Comment:
this isn't sent in gateway v4 or above. you'll get an OP 9 instead. The gateway will not close the connection.